src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2978,15 +2907,10 @@ namespace winrt::TerminalApp::implementation
     // - <none>
     void TerminalPage::_UpdateCommandsForPalette()
     {
-        auto copyOfCommands = _ExpandCommands(_settings.GlobalSettings().ActionMap().NameMap(),
-                                              _settings.ActiveProfiles().GetView(),
-                                              _settings.GlobalSettings().ColorSchemes());
-
-        _recursiveUpdateCommandKeybindingLabels(_settings, copyOfCommands.GetView());
-
         // Update the command palette when settings reload
+        const auto& expanded{ _settings.GlobalSettings().ActionMap().ExpandedCommands() };
```

Comment:
If you make `ExpandedCommands()` return an `IVector` directly you can just pass it to the `CommandPalette()`.